### PR TITLE
Update dependencies that do not break controller-runtime

### PR DESCRIPTION
**Pull request by eqrx**
### Check List Before Merging

- [ ] This PR passes all pre-commit hook validations.
- [ ] This PR is fully tested and regression tests are included.
- [ ] Relevant documentation has been updated.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -18,3 +18,6 @@ bin
 
 # JetBrains IDEs (e.g. GoLand)
 .idea/
+
+# Autogenerated files
+go.work.sum
```
Comment (erdii) [inline .gitignore:23]: This should already be in main. (Is it a no-op to just leave this here?)

File: integration/package-operator/objecttemplate_test.go [hunk 2]
```diff
@@ -44,7 +44,7 @@ func TestObjectTemplate_creationDeletion_packages(t *testing.T) {
 
 	kubernetesKey := "kubernetes"
 	kubernetesPath := ".environment.kubernetes.version"
-	kubernetesValue := "v1.27.1"
+	kubernetesValue := "v1.27.3"
```
Comment (erdii) [inline integration/package-operator/objecttemplate_test.go:47]: Sooo... I don't exactly remember, but did we find out why this change needs to happen?

Comment (eqrx) [reply integration/package-operator/objecttemplate_test.go:47]: because the int test has a specific k8s version hard coded and fixing this is not part of the PR :|

Comment (eqrx) [reply integration/package-operator/objecttemplate_test.go:47]: Issue created

Comment (erdii) [reply integration/package-operator/objecttemplate_test.go:47]: Alrighty... thankersens!
